src/lib/Endpoint/SqlSrvUsers.php

```diff
@@ -0,0 +1,690 @@
+        }
+        $return = array_shift($result);
+        if ($return === null) {
+            throw new Exception\ObjectNotFound('no object found with filter '.$filter);
+        }
+
+        return $this->build($this->prepareRawObject($return), $query);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function create(AttributeMapInterface $map, array $object, bool $simulate = false): ?string
+    {
+        $this->logCreate($object);
+        $loginName = $this->getNameByAttribute($object, self::ATTRLOGINNAME);
+
+        if ($simulate === true) {
+            return null;
+        }
+
+        try {
+            if (isset($object['mechanism']) && $object['mechanism'] === 'windows') {
+                $query = $this->createWindowsLogin($loginName);
+            } else {
+                $query = $this->createLocalLogin($loginName, $object['password'], $object[self::ATTRHASTOCHANGEPWD] ?? true);
+            }
+
+            $this->socket->query($query, $simulate);
+
+            if (isset($object['disabled']) && $object['disabled'] === true) {
+                $this->disableLogin($loginName, $simulate);
+            }
+
+            $principalId = $this->getPrincipalId($loginName);
+        } catch (InvalidQuery $e) {
+            $this->logger->error('failed to create new login user', [
+                'class' => get_class($this),
+                'exception' => $e,
+            ]);
+
+            return null;
+        }
+
+        if (isset($object[self::ATTRSQLNAME]) && $object[self::ATTRSQLNAME] !== '') {
+            try {
+                $sqlName = $this->getNameByAttribute($object, self::ATTRSQLNAME);
+                $this->socket->query($this->createSqlUserQuery($sqlName, $loginName), $simulate);
+
+                if (isset($object[self::ATTRUSERROLES]) && $object[self::ATTRUSERROLES] !== []) {
+                    $this->addRoles($sqlName, $object[self::ATTRUSERROLES], $simulate);
+                }
+            } catch (InvalidQuery $e) {
+                $this->logger->error('failed to create new sql user', [
+                    'class' => get_class($this),
+                    'exception' => $e,
+                ]);
+
+                return null;
+            }
+        } else {
+            $this->logger->info('no sql-user has been created', [
+                'category' => get_class($this),
+            ]);
+        }
+
+        return (string) $principalId;
+    }
+
+    public function delete(AttributeMapInterface $map, array $object, EndpointObjectInterface $endpoint_object, bool $simulate = false): bool
+    {
+        $sqlUserQuery = '';
+        $endpoint_object = $endpoint_object->getData();
+
+        if (isset($endpoint_object[self::ATTRSQLNAME])) {
+            $sqlUserQuery = $this->dropSqlUserQuery($endpoint_object[self::ATTRSQLNAME]);
+        }
+
+        $loginQuery = $this->dropLoginQuery($endpoint_object[self::ATTRLOGINNAME]);
```

Comment:
Same as above, these two operations should probably be a transaction (If possible with login modifications)